pkg/service/restore/worker.go

```diff
@@ -0,0 +1,887 @@
+		}
+
+		return stmt[loc[0]:loc[1]] + " IF NOT EXISTS" + stmt[loc[1]:], nil
+	}
+
+	var views []View
+	for _, ks := range keyspaces {
+		meta, err := w.clusterSession.KeyspaceMetadata(ks)
+		if err != nil {
+			return errors.Wrapf(err, "get keyspace %s metadata", ks)
+		}
+
+		for _, index := range meta.Indexes {
+			if !restoredTables.Has(index.KeyspaceName + "." + index.TableName) {
+				continue
+			}
+			dummyMeta := gocql.KeyspaceMetadata{
+				Indexes: map[string]*gocql.IndexMetadata{index.Name: index},
+			}
+
+			schema, err := dummyMeta.ToCQL()
+			if err != nil {
+				return errors.Wrapf(err, "get index %s.%s create statement", ks, index.Name)
+			}
+
+			// DummyMeta schema consists of create keyspace and create view statements
+			stmt := strings.Split(schema, ";")[1]
+			stmt, err = addIfNotExists(stmt, SecondaryIndex)
+			if err != nil {
+				return err
+			}
+
+			views = append(views, View{
+				Keyspace:   index.KeyspaceName,
+				View:       index.Name,
+				Type:       SecondaryIndex,
+				BaseTable:  index.TableName,
+				CreateStmt: stmt,
+			})
+		}
+
+		for _, view := range meta.Views {
+			if !restoredTables.Has(view.KeyspaceName + "." + view.BaseTableName) {
+				continue
+			}
+			dummyMeta := gocql.KeyspaceMetadata{
+				Views: map[string]*gocql.ViewMetadata{view.ViewName: view},
+			}
+
+			schema, err := dummyMeta.ToCQL()
+			if err != nil {
+				return errors.Wrapf(err, "get view %s.%s create statement", ks, view.ViewName)
+			}
+
+			// DummyMeta schema consists of create keyspace and create view statements
+			stmt := strings.Split(schema, ";")[1]
+			stmt, err = addIfNotExists(stmt, MaterializedView)
+			if err != nil {
+				return err
+			}
+
+			views = append(views, View{
+				Keyspace:   view.KeyspaceName,
+				View:       view.ViewName,
+				Type:       MaterializedView,
+				BaseTable:  view.BaseTableName,
+				CreateStmt: stmt,
+			})
+		}
+	}
+
+	w.logger.Info(ctx, "Initialized views", "views", views)
+	w.run.Views = views
+
+	return nil
+}
+
+// cleanUploadDir deletes all SSTables from host's upload directory except for those present in excluded.
+func (w *worker) cleanUploadDir(ctx context.Context, host, dir string, excluded []string) error {
+	var toBeDeleted []string
+	s := strset.New(excluded...)
+	opts := &scyllaclient.RcloneListDirOpts{FilesOnly: true}
+
+	err := w.client.RcloneListDirIter(ctx, host, dir, opts, func(item *scyllaclient.RcloneListDirItem) {
+		if !s.Has(item.Name) {
+			toBeDeleted = append(toBeDeleted, item.Name)
+		}
+	})
+	if err != nil {
+		return errors.Wrapf(err, "list dir: %s on host: %s", dir, host)
+	}
+
+	if len(toBeDeleted) > 0 {
+		w.logger.Info(ctx, "Delete files from host's upload directory",
+			"host", host,
+			"upload_dir", dir,
+			"files", toBeDeleted,
+		)
+	}
+
+	for _, f := range toBeDeleted {
+		remotePath := path.Join(dir, f)
+		if err := w.client.RcloneDeleteFile(ctx, host, remotePath); err != nil {
```

Comment:
Just out of curiosity. Do we have any RClone method to bulk delete, instead of deleting file by file ?

Reply:
It isn't present in our swagger file, but perhaps Rclone has such method.